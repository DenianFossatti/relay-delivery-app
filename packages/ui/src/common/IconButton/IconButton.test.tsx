import { mockTestID, render, wrapper } from '../../tests';

import { IconButton, IconButtonProps } from '.';

describe(`IconButton`, () => {
  const props: IconButtonProps = {
    accessibilityLabel: 'IconButton',
    icon: 'times',
    testID: 'IconButton',
  };

  const el_container = mockTestID('Button', props.testID!);

  it(`render component #${el_container}`, async () => {
    const { findByTestId, toJSON } = render(<IconButton {...props} />, {
      wrapper,
    });
    const sut = await findByTestId(el_container);
    expect(sut).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
