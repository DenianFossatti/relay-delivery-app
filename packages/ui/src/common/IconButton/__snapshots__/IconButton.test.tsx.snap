// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/IconButton render component #Button_IconButton 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="IconButton"
    accessibilityRole="button"
    accessible={false}
    collapsable={false}
    colorScheme="primary"
    dataSet={{}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHoverIn={[Function]}
    onHoverOut={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 9999,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 10,
      }
    }
    testID="Button_IconButton"
  >
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      test={true}
    >
      <View
        dataSet={{}}
        style={{}}
      >
        <View
          accessible={false}
          testID="Image_Icon"
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
